_posts/en/newsletters/2024-09-27-newsletter.md

```diff
@@ -0,0 +1,267 @@
+  settled (this is new).  Now when Bob receives an endorsed HTLC from
+  Alice:
+
+  - If Bob thinks both Alice and Carol are reliable, he'll forward and
+    endorse the HTLC from Alice to Carol.
+
+  - If Bob thinks only Alice is reliable, he won't forward an endorsed
+    HTLC from Alice.  He'll immediately reject it, allowing the failure
+    to propagate back to the original spender, who can quickly resend
+    using a different route.
+
+  - If Bob thinks only Carol is reliable, he'll accept an endorsed HTLC
+    from Alice when he has extra capacity, but he won't endorse it when
+    forwarding it to Carol.
+
+  Given the change to the proposal, Kirk-Cohen and Shikhelman are
+  planning additional experiments to ensure it works as expected.  They
+  also additionally link to a [mailing list post][posen bidir] by Jim
+  Posen from May 2018 that describes a bidirectional reputation system
+  to prevent jamming attacks (then called _loop attacks_), an example of
+  earlier parallel thinking about solving this problem.
+
+- **Shielded client-side validation (CSV):** Jonas Nick, Liam Eagen, and
+  Robin Linus [posted][nel post] to the Bitcoin-Dev mailing list a
+  [paper][nel paper] about a new [client-side validation][topic
+  client-side validation] protocol.  Client-side validation allows the
+  transfer of tokens to be protected by Bitcoin's proof-of-work without
+  publicly revealing any information about those tokens or transfers.
+  Client-side validation is a key component of protocols such as RGB and
+  Taproot Assets.
+
+  One downside of existing protocols is that the amount of data that
+  needs to be validated by a client when receiving a token is, in the
+  worst case, as large as the history of every transfer of that token
+  and every related token.  In other words, for a set of tokens as
+  frequently exchanged as bitcoins, a client would need to validate a
+  history roughly as large as the entire Bitcoin blockchain.  In
+  addition to the bandwidth cost of transferring that data and the CPU
+  cost of validating it, transferring the full history weakens the
+  privacy of previous receivers of the token.  By comparison, Shielded
+  CSV uses zero-knowledge proofs to allow verification with a fixed
+  amount of resources and without revealing previous transfers.
+
+  Another downside of existing protocols is that each transfer of a
+  token requires including data in a Bitcoin transaction.  Shielded
+  CSV allows multiple transfers to be combined together into the same
+  64-byte update.  This can allow thousands of token transfers to be
+  confirmed each time a new Bitcoin block is found by confirming only a
+  single Bitcoin transaction with an extra 64-byte data push.
+
+  The paper goes into details.  We found particularly interesting the
+  idea of trustlessly bridging bitcoin from the main blockchain to a
+  Shielded CSV (and back) without consensus changes using [BitVM][topic
```

Comment:
Made a small edit that should make this clearer, but just in case it still looks wrong when you're merging, here's a quick explanation: moving BTC to a CSV thing is easy---the client is already processing Bitcoin transactions.  Moving BTC from a CSV thing back to mainnet is hard---to prove the transfer is authorized, you typically need operations not available in Script.  The paper authors claim that BitVM can work around this.

In other words, bi-directional bridging would use BitVM.

To go into unnecessary detail, I suspect there's some additional constraints that the authors haven't mentioned (or that I didn't see in my quick skim).  BitVM can verify execution of arbitrary programs---but it depends on at least one honest user independently performing verification and being able to submit a fraud proof before the money is withdrawn from the BitVM contract.  In other words, anyone with BTC in a Shielded CSV would need to come online periodically to check whether someone was trying to steal their BTC (similar to LN).

Reply:
Thanks for clarifying. I could see how someone could have read the original "without consensus changes using BitVM" as "without consensus changes that use BitVM"